modules/template/template.go

```diff
@@ -104,6 +105,15 @@ func NewFuncMap() []template.FuncMap {
 		"ThemeColorMetaTag": func() string {
 			return setting.ThemeColorMetaTag
 		},
+		"TabSizeClass": func(ec *editorconfig.Editorconfig, filename string) string {
+			if ec != nil {
+				def := ec.GetDefinitionForFilename(filename)
+				if def.TabWidth > 0 {
+					return fmt.Sprintf("tab-size-%d", def.TabWidth)
+				}
+			}
+			return "tab-size-8"
```

Comment:
the argument for 8 is that, while archaic, it's the default on Github too.
the argument for 4 is that it's more common.

there could also be made an argument for 2 spaces, it's predominant in a few languages, so ultimatively, there should be an option. maybe in `app.ini` as a start, but I don't see it as a blocker on this PR.